static/js/settings_users.js

```diff
@@ -505,19 +506,27 @@ function handle_reactivation($tbody, $status_field) {
         const $button_elem = $(e.target);
         const $row = $button_elem.closest(".user_row");
         const user_id = Number.parseInt($row.attr("data-user-id"), 10);
-        const url = "/json/users/" + encodeURIComponent(user_id) + "/reactivate";
-        const data = {};
 
-        const opts = {
-            success_continuation() {
-                update_view_on_reactivate($row);
-            },
-            error_continuation(xhr) {
-                ui_report.generic_row_button_error(xhr, $button_elem);
-            },
-        };
+        function handle_confirm() {
+            const $row = get_user_info_row(user_id);
+            const $row_reactivate_button = $row.find("button.reactivate");
+            $row_reactivate_button.prop("disabled", true).text($t({defaultMessage: "Working…"}));
+            const url = "/json/users/" + encodeURIComponent(user_id) + "/reactivate";
+            const data = {};
+
+            const opts = {
+                success_continuation() {
+                    update_view_on_reactivate($row);
+                },
+                error_continuation(xhr) {
+                    ui_report.generic_row_button_error(xhr, $button_elem);
+                },
```

Comment:
Are these success/fail handlers correct? I would expect us to have any errors appear in the modal itself, not in the tale.

Reply:
I think these are correct for the table and buttons.

I would test them for modal too.

Reply:
These lines would be enough?

```
    success() {
        dialog_widget.close_modal();
    },
    error(xhr) {
        ui_report.error($t_html({defaultMessage: "Failed"}), xhr, $("#dialog_error"));
        dialog_widget.hide_dialog_spinner();
```

Reply:
> Are these success/fail handlers correct? I would expect us to have any errors appear in the modal itself, not in the table.
> 
> Might be good to make a screencast where you modify the backend to return an error on the relevant API request to show how the error handling works.

Currently we show error in the table (where we show the `Saving` indicator) when deactivating the modal from the settings page, and we show the error in modal in case we deactivate from the `Manage user` modal opened directly from user profile popover.

Reply:
When I tested for errors.
<details>
<summary>When Invalid User</summary>

![when_invalid_user_id](https://user-images.githubusercontent.com/41695888/159294047-ed40ea37-c912-49ea-b0df-45bf1a466a54.png)
</details>
<details>
<summary>When Not Allow Deactivated</summary>

![when_not_allow_deactivated](https://user-images.githubusercontent.com/41695888/159294049-3a324fe3-a25b-47ac-a81f-bd6f0809855e.png)
</details>
<details>
<summary>When Insufficient Permission</summary>

![when_insufficient_permission](https://user-images.githubusercontent.com/41695888/159294041-2975c9d4-9579-41cf-b005-283e12a9d39d.png)
</details>

https://github.com/zulip/zulip/blob/af5d0d6f5e5444332f9f8e565d97f4acdceaa72f/zerver/lib/users.py#L294-L295

https://github.com/zulip/zulip/blob/af5d0d6f5e5444332f9f8e565d97f4acdceaa72f/zerver/lib/users.py#L267-L276

Reply:
I think we should show error only in one place, and I think at present it should be at the top where we show error in `Users` table on deactivating, not where the reactivate button is present.

Reply:
Then should we disable the button or just show it as it is?

Reply:
Disable the button and change the text to `Working` till the server returns response (this part is same as we do for deactivating user now in `Users` list)  showing the error on the top and then re-enabling the button with icon and as discussed below